### PR TITLE
Pf/third party sanity

### PR DESCRIPTION
**Pull request by pfinashx**
Added tests for third-party integration, rebased on develop

### CODE REVIEW

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Remove the `tests/data/mock_datasets/glue/glue_data/MNLI/cached_dev_roberta_mnli_128_mnli` and other `cached*` files.

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

File: tests/test_sanity_third_party.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                       cwd=self.MMDET_PATH)
+        subprocess.run("{}; pip install \"git+https://github.com/cocodataset/cocoapi.git#subdirectory=PythonAPI\""
+                       .format(self.activate_venv),
+                       check=True, shell=True, cwd=self.MMDET_PATH)
+        subprocess.run("{}; pip install -v -e .".format(self.activate_venv), check=True, shell=True,
+                       cwd=self.MMDET_PATH)
+        subprocess.run("{}; cp {} .".format(self.activate_venv, self.PATH_TO_PATCH), check=True, shell=True,
+                       cwd=self.MMDET_PATH)
+        subprocess.run("{}; git apply 0001-Modifications-for-NNCF-usage.patch".format(self.activate_venv),
+                       check=True, shell=True, cwd=self.MMDET_PATH)
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/test_sanity_third_party.py]: Done.
Precommit: http://134.191.240.124/job/nncf/job/precommit/job/manual_precommit/76/

File: tests/test_sanity_third_party.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        subprocess.run("{} && pip install boto3".format(self.activate_venv), check=True, shell=True,
+                       cwd=self.TRANS_PATH)
+        subprocess.run("{} && git checkout {}".format(self.activate_venv, TRANSFORMERS_COMMIT), check=True, shell=True,
+                       cwd=self.TRANS_PATH)
+        subprocess.run(
+            "{} && {}/bin/python setup.py develop".format(self.activate_venv, self.VENV_TRANS_PATH), check=True,
+            shell=True, cwd=PROJECT_ROOT)
+        subprocess.run("{} && cp {} .".format(self.activate_venv, self.PATH_TO_PATCH), check=True, shell=True,
+                       cwd=self.TRANS_PATH)
+        subprocess.run("{} && git apply 0001-Modifications-for-NNCF-usage.patch".format(self.activate_venv),
```
[the remark this answers is not in the document]
Comment (pfinashx) [reply tests/test_sanity_third_party.py]: Done.
Precommit: http://134.191.240.124/job/nncf/job/precommit/job/manual_precommit/76/

**Comment by vshampor** [comment]
Jenkins please retry a build

**Pull request merged**